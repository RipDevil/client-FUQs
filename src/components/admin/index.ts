export * from './grips/users';
export * from './grips/fuqs';
export * from './portal';
