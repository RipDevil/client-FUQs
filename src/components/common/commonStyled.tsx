import styled from 'styled-components';

export const StyledSpinner = {
  SpinnerDiv: styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    z-index: 1000;
    margin: -50px 0px 0px -50px;
    max-width: 140px;
  `,
  BackgroundDiv: styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 999;
    background-color: rgb(255, 255, 255);
  `,
  TextDiv: styled.div`
    text-align: center;
    margin-top: 1rem;
  `,
};
