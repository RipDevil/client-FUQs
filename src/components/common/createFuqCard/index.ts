export { CreateFuqCard } from './create-fuq-card';
