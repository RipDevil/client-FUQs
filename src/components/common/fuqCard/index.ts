export { FuqCard } from './fuq-card';
