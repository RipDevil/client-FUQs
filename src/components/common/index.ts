export { Credits } from './credits';
export { FuqCard } from './fuqCard';
export { Spinner } from './spinner';
export { ToggleIcon } from './toggleIcon';
export { CreateFuqCard } from './createFuqCard';
export { InfluencerBadge } from './influencerBadge';
