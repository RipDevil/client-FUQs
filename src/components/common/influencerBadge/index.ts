export { InfluencerBadge } from './influencer-badge';
