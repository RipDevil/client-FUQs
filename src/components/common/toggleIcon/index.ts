export { ToggleIcon } from './toggle-icon';
