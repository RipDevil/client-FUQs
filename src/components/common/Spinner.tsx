import * as React from 'react';
import { StyledSpinner as S } from './commonStyled';

export interface SpinnerProps {
  text?: string;
  transparent?: boolean;
}

const Spinner: React.FC<SpinnerProps> = ({
  text = '',
  transparent = false, // TODO: implement this stuff
}) => {
  return (
    <>
      <S.BackgroundDiv />
      <S.SpinnerDiv>
        <div>
          <i className='fas fa-10x fa-spin fa-cog' />
        </div>
        {text && (
          <S.TextDiv>
            <h3>{text}</h3>
          </S.TextDiv>
        )}
      </S.SpinnerDiv>
    </>
  );
};

export default Spinner;
