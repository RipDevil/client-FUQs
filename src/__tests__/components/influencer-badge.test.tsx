import { render, waitFor, cleanup, fireEvent, screen } from '@testing-library/react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { InfluencerBadge } from 'components/common';

afterEach(() => {
  cleanup();
});

describe('Influencer badge tests', () => {
  it('Should render with no errors', () => {
    const { container } = render(
      <Router history={createMemoryHistory({ initialEntries: ['/fuq'] })}>
        <Route path={'/fuq'}>
          <InfluencerBadge url="/" />
        </Route>
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should contain link which is passed via props', () => {
    const { container } = render(
      <Router history={createMemoryHistory({ initialEntries: ['/fuq'] })}>
        <Route path={'/fuq'}>
          <InfluencerBadge url="/test-link" />
        </Route>
        <Route path={'/test-link'}>
          <h1>Test</h1>
        </Route>
      </Router>,
    );
    const linkElement = container.querySelector('a[href="/test-link"]');

    expect(linkElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('Should not render if no url is passed', () => {
    const { container } = render(<InfluencerBadge />);
    expect(container).toMatchSnapshot();
  });

  it('Should redirect on the Link click', async () => {
    const { container, getByText } = render(
      <Router history={createMemoryHistory({ initialEntries: ['/fuq'] })}>
        <Route path={'/fuq'}>
          <InfluencerBadge url="/test-link" />
        </Route>
        <Route path={'/test-link'}>
          <h1>Test</h1>
        </Route>
      </Router>,
    );

    const linkElement = container.querySelector('a[href="/test-link"]');
    expect(linkElement).toBeInTheDocument();

    // ugly?
    expect(linkElement).not.toBeNull();
    if (linkElement) {
      fireEvent.click(linkElement);

      const isAnotherPage = await waitFor(() => {
        return getByText(/Test/i);
      });
      expect(isAnotherPage).toBeTruthy();
    }
  });
});
