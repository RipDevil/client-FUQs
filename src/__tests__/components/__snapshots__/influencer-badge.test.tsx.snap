// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Influencer badge tests Should contain link which is passed via props 1`] = `
.c0 {
  font-size: small;
  font-style: italic;
}

<div>
  <span
    class="ant-typography ant-typography-secondary c0"
  >
    this FUQ has been influenced by 
    <a
      href="/test-link"
    >
      this FUQ
    </a>
  </span>
</div>
`;

exports[`Influencer badge tests Should not render if no url is passed 1`] = `<div />`;

exports[`Influencer badge tests Should render with no errors 1`] = `
.c0 {
  font-size: small;
  font-style: italic;
}

<div>
  <span
    class="ant-typography ant-typography-secondary c0"
  >
    this FUQ has been influenced by 
    <a
      href="/"
    >
      this FUQ
    </a>
  </span>
</div>
`;
